### PR TITLE
Kaggle display fix

### PR DESCRIPTION
**Pull request by ItayGabbay**
#### What does this implement/fix? Explain your changes.
Fixing the display on Kaggle.

---
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/deepchecks/deepchecks/blob/main/CONTRIBUTING.rst

### CODE REVIEW

**Review by matanper — APPROVED** [review]

File: deepchecks/core/display.py [hunk 1]
```diff
@@ -92,7 +92,9 @@ def _repr_html_(self):
 
             return TempSphinx()
 
-        if is_colab_env() and as_widget is True:
+        if is_kaggle_env():
```
Comment (yromanyshyn) [inline deepchecks/core/display.py:95]: you need then change it to this

```
if is_kaggle_env() and is_databricks_env() and is_sagemaker_env():
    self.show_in_iframe(as_widget=as_widget, unique_id=unique_id, **kwargs)
```

**Review by yromanyshyn — CHANGES_REQUESTED** [review]

File: deepchecks/core/display.py (hunk 1 above)
Comment (ItayGabbay) [reply deepchecks/core/display.py:95]: Do we have a mechanism to check if we're on sagemaker?

Comment (yromanyshyn) [reply deepchecks/core/display.py:95]: plus, I think it is worth showing a warning, something like 
`this envs do not support ipywidgets (or have troubles with thems) therefore we will use iframe`

**Review by yromanyshyn — APPROVED** [review]

File: deepchecks/core/display.py (hunk 1 above)
Comment (yromanyshyn) [reply deepchecks/core/display.py]: sorry, did not see you prev comment, no we do not have

**Pull request merged**